### ISSUE TITLE
Configurable placeholder refresh rates per layer

### ISSUE THREAD

**Issue by Complex1t** [Ehhthan/HappyHud-Issues#33]
Currently, there is only one way to change how frequently HappyHUD checks for all placeholders, using the `check-update` option in config.yml. However, it would be beneficial to have the option to tweak placeholder refresh rates per layer. This can result in performance improvements as users could choose certain layers to frequently update (such as health) and other layers to not update as frequently (time of day, votes, etc.)

**Comment by Ehhthan** [comment]
I don't really believe placeholders take a significant amount of performance to check for an update. The way it currently works is an update is called for all layers that have a placeholder. Once that is sent, the placeholders are parsed and compared with the cached component. If a change is detected the cached value is replaced and the new component is sent. In theory I could cache the text instead of a component, but not sure how much of an improvement it would be. Let me know what you think.

**Comment by Complex1t** [comment]
Hmm okay. I don't believe HappyHUD itself refreshing placeholders are using a lot of resources, but it is actually the cascading _result_ of HappyHUD refreshing placeholders/layers that could be intensive.

Here is a snip from a spark report.
https://i.imgur.com/Lmzcrnx.png

Out of the total 1.09% server thread use, 41% of the profile was spent on FunctionalActiveLayer.update(). I'd assume this is what updates the layers. Out of this, 17% was used on deserialize() and 18% was used in parsing placeholders.  With this, plugins would have to respond to the placeholder request, which would further take more resources.

If there was a way to configure the refresh rates of each layers, not all of them would have to be updated frequently at once. Other plugins wouldn't need to respond to the frequent placeholder requests as frequently as well, and therefore HappyHUD wouldn't need to deserialize for MiniMessage as frequently and as much either.

Another method is sendActionBar, taking a big 47% of the total server thread use by HappyHUD. I don't know why creating and sending an action bar would be so intensive... but I'm assuming nothing can be done about this as no matter the varying refresh rates of layers, HappyHUD would still need to constantly send action bars to players to keep the HUD active.

Now I should note that you are the developer and I have no idea of the internals of HappyHUD... I'm just hoping it helps you haha. Also, there have been some great improvements lately in both performance and bug fixes. Thank you.

**Comment by Ehhthan** [comment]
Yep, your analysis definitely helps targeting what can and cannot be improved. I have plans to improve how bars are built and cached. This will hopefully bring down the timings even lower. As for the action bar, it seems like it's just something that takes a bit more than usual. The smaller the component the more performant so decreasing that a bit could show some improvement. It also does not help that I have to run that sync. I'll do some testing with just directly sending the component vs an adventure audience.
